### ISSUE TITLE
Windows: Tests in the root of the workspace are broken in Bazel 0.12.0

### ISSUE THREAD

**Issue by jasharpe** [bazelbuild/bazel#5079]
### Description of the problem / feature request:

All tests in the root of the workspace are broken in Bazel 0.12.0, failing with

```
external/bazel_tools/tools/test/test-setup.sh: line 231: : command not found
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Create a repo on a Windows machine with an empty WORKSPACE and the contents of this gist: https://gist.github.com/jasharpe/ca1cdbeb293d0d1cd1d552dbabee0bb5

In the root of the workspace, with Bazel 0.12.0, run:

```
bazel clean; bazel test :sh_test --test_output=errors
```

Actual:

```
==================== Test output for //:sh_test:
external/bazel_tools/tools/test/test-setup.sh: line 231: : command not found
================================================================================
```

Expected: Test should pass.

### What operating system are you running Bazel on?

Windows Server 2016

### What's the output of `bazel info release`?

release 0.12.0

### If `bazel info release` returns "development version" or "(@non-git)", tell us how you built Bazel.

N/A

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

N/A

###  Have you found anything relevant by searching the web?

No.

### Any other information, logs, or outputs that you want to share?

This is broken because for a test in the root of the workspace, Bazel is looking in the runfiles manifest for:

```
__main__/./sh_test.exe
```

(Note the "/./" in the middle! That's the problem.)

But the manifest has an entry  that looks like this:

```
...
__main__/sh_test.exe ...
...
```

(Note no "/./" in the middle.)

I am pretty sure this was working in Bazel 0.11.

**Comment by jasharpe** [comment]
@laszlocsomor @meteorcloudy FYI

**Comment by jasharpe** [comment]
BTW this is broken in the same way at HEAD.

**Comment by jasharpe** [comment]
Confirmed that this is *not broken* on Bazel 0.11.1. The exact same test command passes.

**Comment by laszlocsomor** [comment]
@jasharpe , thanks for a detailed bug report! I'll take a look.

**Comment by laszlocsomor** [comment]
Culprit is eb067ea88749a5635cc8ee8954cde2b767f1eb61.

**Comment by laszlocsomor** [comment]
This is blocking https://github.com/bazelbuild/bazel/issues/4583

**Comment by laszlocsomor** [comment]
We'll cherry-pick a fix into 0.13.0 . Closing this issue.
